File: programmers/Level 2/the_biggest_number.cpp
// programmers Level 2: 가장 큰 수
#include <string>
#include <vector>
#include <algorithm>
#include <iostream>
using namespace std;

bool cmp(string &a, string &b){
    return a + b > b + a;
}

string solution(vector<int> numbers) {
    string answer = "";
    vector<string> v;
    
    for(int i = 0 ; i < numbers.size() ; i++) v.push_back(to_string(numbers[i]));
    sort(v.begin(), v.end(), cmp);

    for(int i = 0 ; i < v.size() ; i++) answer += v[i];
    answer = answer.front() == '0' ? "0" : answer;
    
    return answer;
}

File: programmers/Level 2/the_biggest_number.js
/*
programmers Level 2: 가장 큰 수
2021-12-27
*/
function solution(numbers) {
    let answer = numbers.sort((a, b) => `${b}${a}` - `${a}${b}`).join('');
    answer = answer[0] === "0" ? "0" : answer;

    return answer;
}